### PR TITLE
Feature/sg 431 check classes

### PR DESCRIPTION
**Pull request by BloodAxe**
Fix https://deci.atlassian.net/browse/SG-431

### CODE REVIEW

File: src/super_gradients/training/datasets/detection_datasets/coco_detection.py [hunk 1]
```diff
@@ -46,14 +48,27 @@ def __init__(
         kwargs["all_classes_list"] = kwargs.get("all_classes_list") or COCO_DETECTION_CLASSES_LIST
         super().__init__(*args, **kwargs)
 
+        if len(self.original_classes) != len(self.all_classes_list):
```
Comment (ofrimasad) [inline src/super_gradients/training/datasets/detection_datasets/coco_detection.py:51]: don't we want to check this before calling `super().__init__(...)`

**Review by ofrimasad — APPROVED** [review]
LGTM... just a minor question/comment

File: src/super_gradients/training/datasets/detection_datasets/coco_detection.py (hunk 1 above)
Comment (BloodAxe) [reply src/super_gradients/training/datasets/detection_datasets/coco_detection.py:51]: We don't, as we rely on value `self.original_classes` which is initialized from `_setup_data_source` which is called from base c'tor. 
The dataset initialization is quite puzzled and could be refactored a bit to make it less tangled though.

**Review by ofrimasad — APPROVED** [review]
LGTM

**Pull request merged**